packages/dds/tree/src/feature-libraries/modular-schema/modularChangeTypes.ts

```diff
@@ -22,15 +23,16 @@ export interface ModularChangeset extends HasFieldChanges {
 	 * The numerically highest `ChangesetLocalId` used in this changeset.
 	 * If undefined then this changeset contains no IDs.
 	 */
-	maxId?: ChangesetLocalId;
+	readonly maxId?: ChangesetLocalId;
 	/**
 	 * The revisions included in this changeset, ordered temporally (oldest to newest).
 	 * Undefined for anonymous changesets.
 	 * Should never be empty.
 	 */
 	readonly revisions?: readonly RevisionInfo[];
-	fieldChanges: FieldChangeMap;
-	constraintViolationCount?: number;
+	readonly fieldChanges: FieldChangeMap;
```

Comment:
Would we end up with less code duplication if we got rid of `fieldChanges`, and instead had a `rootChanges?: NodeId` that would contain those `fieldChanges`? (The term "root" in "rootChanges" is a little confusing here: it refers to the abstract root that is always present and whose fields are the detached fields).

Reply:
I'm not convinced this would simplify the code much at all, and I'm not sure it would make sense to have a `NodeChangeset` for the root if we don't really consider it a node.